### PR TITLE
Add a user friendly error message for http exceptions

### PR DESCRIPTION
**Pull request by grandbora**
**What this PR does / why we need it**: 
Adds a user friendly error message to http exceptions that may happen when sending requests to the predictor node.

Example error log:
```
2024-12-12 12:12:12.333 37708 kserve ERROR [_http_predict():279] Could not send a request to predictor at url https://predictor/generate with headers {'Content-Type': 'application/json', 'x-request-id': 'xxxxx'} and with data b'{"inputs":"generate prompt text.","parameters":{"max_new_tokens":128,"temperature":0.9,"top_p":0.9,"top_k":50,"seed":12345,"details":true,"do_sample":true}}' due to exception [Errno 8] nodename nor servname provided, or not known
```

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:
Tested locally. Spun up a transformer and pointed it to a wrong url. Generated the log message above.

**Special notes for your reviewer**:
No image changes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

Not worth of a release note.

### CODE REVIEW

**Comment by grandbora** [comment]
cc @yuzisun @cmaddalozzo

File: python/kserve/kserve/model.py [hunk 1]
```diff
@@ -317,9 +317,20 @@ async def _http_predict(
...
+
+        try:
+            response = await self._http_client.post(
+                predict_url, timeout=self.timeout, headers=predict_headers, content=data
+            )
+        except Exception as exc:
+            logger.error(
+                f"Could not send a request to predictor at url {predict_url} "
+                f"with headers {predict_headers} "
+                f"and with data {data} "
```
Comment (terrytangyuan) [inline python/kserve/kserve/model.py]: What if there's sensitive data?

Comment (grandbora) [reply python/kserve/kserve/model.py]: Hi Yuan, I had this debate internally, thanks for raising it. The companies that need to protect their logs, do not rely on the libraries they use not to spew any data. They capture logs, scrub and obfuscate them, or hide them all together.

I think in this regard Kserve's priority should be providing a good developer experience. Its concern on privacy should not go beyond ensuring that logs are written only to `stdout` or `stderr` and nowehere else. This way the users who are interested in processing the logs, can do so safely.

What do you think?

**Review by terrytangyuan — CHANGES_REQUESTED** [review]
See comment

**Comment by grandbora** [comment]
@terrytangyuan removed logging for potentially sensitive data.

**Review by terrytangyuan — APPROVED** [review]
/lgtm

**Comment by terrytangyuan** [comment]
Can you sign DCO?

**Comment by grandbora** [comment]
Yes, signing the DCO in a minute.

**Review by spolti — APPROVED** [review]
Nice addition, thanks!

**Comment by grandbora** [comment]
@terrytangyuan signed the DCO. Thank you.

**Comment by grandbora** [comment]
@jinchihe would you mind approving this pr?

**Comment by grandbora** [comment]
@jinchihe @terrytangyuan @yuzisun is there anything else needs to be done for this pr?

**Comment by grandbora** [comment]
@yuzisun @jinchihe I'd appreciate 👀 on this.

**Comment by terrytangyuan** [comment]
@grandbora You'll have to be a little more patient since the only active approver is @yuzisun right now.

**Pull request merged**